keps/sig-scheduling/4816-dra-prioritized-list/README.md

```diff
@@ -255,17 +255,40 @@ nitty-gritty.
 The `ResourceClaim` object contains a `DeviceClaim`, which in turn contains a
 list of `DeviceRequest` objects. This allows the user to allocate different
 types of devices for the same claim, and apply constraints and configuration
-across those different requests.
+across those different requests. These changes allow some flexibility for the
+user to create, say, a "gpu" request, but allow it to be satisfied by one of
+several models of GPU.
 
-The proposal adds a new field to the `DeviceRequest`, called `FirstAvailableOf`
-which will contain an ordered list of `DeviceRequest` objects. In order to
+To avoid introducing a breaking API change and to ensure round-tripping
+between the APIs, the `v1beta1` API will remain backward compatible 
+(only extended) and the final change will be made in `v1beta2`.
+
+The `v1beta1` API will be extended with a new field called `FirstAvailable`
+which will contain an ordered list of `DeviceSubRequest` objects. In order to
 satisfy the main (containing) request, exactly one of the requests listed in
-`FirstAvailableOf` must be satisfied. The order listed is considered a priority
+`FirstAvailable` must be satisfied. The order listed is considered a priority
 order, such that the scheduler will only try to use the second item in the list
-if it is unable to satisfy the first item, and so on.
-
-This allows some flexibility for the user to create, say, a "gpu" request, but
-allow it to be satisfied by one of several models of GPU.
+if it is unable to satisfy the first item, and so on. This extension to the API
+is not breaking, but it makes for a somewhat awkward API where many fields on
+the `DeviceRequest` (in fact all existing fields except `Name`) must not be
+set whenever `FirstAvailable` is set. To avoid this API in the long term, we are
+adding the `v1beta2` API described next.
+
+For the `v1beta2` API, the `DeviceRequest` object will be restructured
+so that there will be the `Name` field just like today, and for supporting
+either a single request or a prioritized list of subrequests, there will
+be separate fields:
+
+* `Exactly` for a request of type `SpecificDeviceRequest` without alternatives.
+`SpecificDeviceRequest` will include all the fields that exists on the
+`DeviceRequest` type in `v1beta1`, except the `Name` field.
+* `FirstAvailable` for providing a prioritized list of requests, each of
+type `DeviceSubRequest`. The `DeviceSubRequest` type is similar to
+`SpecificDeviceRequest`, except for the `AdminAccess` field that is not
+available when providing multiple alternatives. The list provided in the
+`FirstAvailable` field is considered a priority order, such that the
+scheduler will only try to use the second item in the list if it is unable
```

Comment:
"try to use" will require a more precise definition. IIUC, a Node could now pass the Filter if any of the `FirstAvailable` can be met. And then we would have to do something in the Score phase to prefer the node that satisfies one of the earlier items.

Not saying it has to be fully explained here, but it probably has to be explained in the API comments and the documentation.

Reply:
I've updated the language here and also the comments in the code snippets. I've tried to make the behavior clearer and also point out the limitations here since this change does not add scoring (which is being discussed in https://github.com/kubernetes/enhancements/issues/4970)